iocore/net/SSLClientUtils.cc

```diff
@@ -54,21 +55,23 @@ verify_callback(int preverify_ok, X509_STORE_CTX *ctx)
    */
   ssl                      = static_cast<SSL *>(X509_STORE_CTX_get_ex_data(ctx, SSL_get_ex_data_X509_STORE_CTX_idx()));
   SSLNetVConnection *netvc = SSLNetVCAccess(ssl);
+  bool enforce_mode        = (netvc && netvc->options.clientVerificationFlag == (uint8_t)YamlSNIConfig::Level::STRICT);
   if (!preverify_ok) {
     // Don't bother to check the hostname if we failed openssl's verification
     SSLDebug("verify error:num=%d:%s:depth=%d", err, X509_verify_cert_error_string(err), depth);
-    if (netvc && netvc->options.clientVerificationFlag == 2) {
-      if (netvc->options.sni_servername) {
-        Warning("Hostname verification failed for (%s) but still continuing with the connection establishment",
-                netvc->options.sni_servername.get());
-      } else {
-        char buff[INET6_ADDRSTRLEN];
-        ats_ip_ntop(netvc->get_remote_addr(), buff, INET6_ADDRSTRLEN);
-        Warning("Server certificate verification failed for %s but still continuing with the connection establishment", buff);
-      }
-      return 1;
+    const char *sni_name;
+    char buff[INET6_ADDRSTRLEN];
+    ats_ip_ntop(netvc->get_remote_addr(), buff, INET6_ADDRSTRLEN);
```

Comment:
Should keep the `ats_ip_ntop` inside the `else` clause and just move the buffer declaration.

Reply:
We print the value of buff in either case since we are showing the IP address of the server in the warning message.